src/commonMain/kotlin/com/github/quillraven/fleks/world.kt

```diff
@@ -177,6 +177,60 @@ class WorldConfiguration(@PublishedApi internal val world: World) {
             // instead of resizing every time a system gets added to the configuration
             world.systems = it.systems.toTypedArray()
         }
+
+        setUpAggregatedFamilyHooks()
+    }
+
+    /**
+     * Extend [Family.addHook] and [Family.removeHook] with
+     * other objects that needed to triggered by the hooks.
+     */
+    private fun setUpAggregatedFamilyHooks() {
+
+        // Validate systems against illegal interfaces.
+        world.systems.forEach { system ->
+            // FamilyOnAdd and FamilyOnRemove interfaces are only meant to be used by IteratingSystem.
+            if (system !is IteratingSystem) {
+
+                if (system is IteratingSystem.FamilyOnAdd)
```

Comment:
please add '{' braces to your ifs. I only skip them if it is a short one-liner if.